addons/im_livechat/controllers/main.py

```diff
@@ -125,38 +124,41 @@ def get_session(self, channel_id, anonymous_name, previous_operator_id=None, cha
                 }
                 store.add(chatbot_script)
                 store.add(welcome_steps)
-            operator = request.env["res.partner"].sudo().browse(channel_vals["livechat_operator_id"])
+            operator_partner = agent.partner_id if agent else chatbot_script.operator_partner_id
             channel_info = {
                 "fetchChannelInfoState": "fetched",
                 "id": channel_id,
                 "isLoaded": True,
                 "livechat_operator_id": Store.One(
-                    operator, self.env["discuss.channel"]._store_livechat_operator_id_fields(),
+                    operator_partner, self.env["discuss.channel"]._store_livechat_operator_id_fields(),
                 ),
-                "name": channel_vals["name"],
                 "scrollUnread": False,
                 "channel_type": "livechat",
                 "chatbot": chatbot_data,
             }
             store.add_model_values("discuss.channel", channel_info)
         else:
+            if request.env.user._is_public():
+                guest = guest.sudo()._get_or_create_guest(
+                    guest_name=self._get_guest_name(),
+                    country_code=request.geoip.country_code,
+                    timezone=request.env["mail.guest"]._get_timezone_from_request(request),
+                )
+                livechat_channel = livechat_channel.with_context(guest=guest)
+            channel_vals = livechat_channel._get_livechat_discuss_channel_vals(
+                chatbot_script=chatbot_script, agent=agent
+            )
+            lang = request.env["res.lang"].search(
+                [("code", "=", request.cookies.get("frontend_lang"))]
+            )
+            channel_vals.update({"country_id": country.id, "livechat_lang_id": lang.id})
             channel = request.env['discuss.channel'].with_context(
                 mail_create_nosubscribe=False,
```

Comment:
Isn't `mail_create_nosubscribe` obsolete and wrong code?
We always force `mail_create_nosubscribe=True` in create override of discuss.channel, as we should never have followers in channels.